core/src/avm1/globals/movie_clip.rs

```diff
@@ -1418,7 +1431,23 @@ fn get_bounds<'gc>(
             // the final matrix, but this matches Flash's behavior.
             let to_global_matrix = movie_clip.local_to_global_matrix();
             let to_target_matrix = target.global_to_local_matrix().unwrap_or_default();
-            to_target_matrix * to_global_matrix * bounds
+            let target_bounds = to_target_matrix * to_global_matrix * bounds.clone();
+
+            // If the bounds are invalid, the target space is identical to the origin space and
+            // use_new_invalid_bounds_value is true, the returned bounds use a specific invalid value.
+            if *activation.context.use_new_invalid_bounds_value
```

Comment:
Does this really need to be a field in `UpdateContext`? Could you instead perform the check

```rust
            if activation.swf_version() >= 8
                || movie_clip.get_oldest_parent_swf_version(&mut activation.context) >= 8
```

after you've found that `bounds` and `target_bounds` are invalid?

Reply:
Sadly, it needs to be. The behaviour in Flash is a bit weird: As soon as the call returned `X.4` for one time, it will always return `X.4` for invalid bounds with a different target. It is an actual internal state changing irreversibly.

So e.g. there can be the following scenario (that's also tested in the tests):
- Parent MovieClip with SWF version 7 calls `getBounds` => the check is `false` and it returns `X.35`
- Then, child movie clip with SWF version 8 is loaded and calls `getBounds` => the check is `true` and it returns `X.4`
- Parent MovieClip then calls `getBounds` again => even if the check is still `false`, it returns `X.4` (because the state changed with the second call).

Because this behaviour is so specific and complex, I added quite some `movieclip_invalid_bounds` tests making sure Ruffle exactly behaves like this.